spec/lib/kafka/avro_producer_spec.rb

```diff
@@ -44,25 +46,57 @@
   end
 
   context 'producing a message successfully' do
+    let(:topic1_payload_value) { "\x00\x00\x00\x00\x05\x02\x06key\nvalue\x00" }
+
     before do
       Kafka::ProducerManager.instance.send(:setup_producer)
-      allow(avro_producer).to receive(:get_schema).and_return(schema)
     end
 
     after do
       # reset the client after each test
       avro_producer.producer.client.reset
     end
 
-    it 'produces a message to the specified topic' do
-      avro_producer.produce('topic-1', valid_payload)
-      avro_producer.produce('topic-1', valid_payload)
-      avro_producer.produce('topic-2', valid_payload)
+    context 'with dynamic schema registry retrieval' do
+      context 'of an existing schema' do
+        it 'produces a message to the specified topic' do
+          VCR.use_cassette('kafka/topics') do
+            avro_producer.produce('topic-1', valid_payload)
+            avro_producer.produce('topic-2', valid_payload)
+            expect(avro_producer.producer.client.messages.length).to eq(2)
+            topic_1_messages = avro_producer.producer.client.messages_for('topic-1')
+            expect(topic_1_messages.length).to eq(1)
+            expect(topic_1_messages[0][:payload]).to be_a(String)
+            expect(topic_1_messages[0][:payload]).to eq(topic1_payload_value)
+          end
+        end
+      end
+
+      context 'of an non-existing schema' do
+        it 'raises approriate error' do
+          VCR.use_cassette('kafka/topics404') do
+            expect do
+              avro_producer.produce('topic-999', valid_payload)
+            end.to raise_error(Faraday::ResourceNotFound)
+          end
+        end
+      end
+    end
+
+    context 'with hardcoded schema registry retrieval' do
+      before do
+        allow(Flipper).to receive(:enabled?).with(:kafka_producer_fetch_schema_dynamically).and_return(false)
+        allow(File).to receive(:read).and_return(schema_file)
```

Comment:
This is a workaround to a problem i was having where only on the CI pipeline was it not finding the hardcoded schema file in the producer but it can find it in the spec apparently.